cpp/include/cudf/detail/utilities/vector_factories.hpp

```diff
@@ -439,6 +441,40 @@ thrust::host_vector<typename Container::value_type> make_host_vector_sync(
   return make_host_vector_sync(device_span<typename Container::value_type const>{c}, stream);
 }
 
+/**
+ * @brief Asynchronously construct a `host_vector` of the given size
+ *
+ * @note This function does not synchronize `stream`.
+ *
+ * @tparam T The type of the vector data
+ * @param size The number of elements in the created vector
+ * @param stream The stream on which to allocate memory
+ * @return A host_vector of the given size
+ */
+template <typename T>
+host_vector<T> make_pinned_vector_async(size_t size, rmm::cuda_stream_view stream)
+{
+  return host_vector<T>(size, {cudf::get_pinned_memory_resource(), stream});
+}
+
+/**
+ * @brief Synchronously construct a `cudf::detail::host_vector` of the given size
```

Comment:
same here, re: comment.

Reply:
added minimalist comments, let me know if it should be more elaborate